bin/varnishd/cache/cache_obj.c

```diff
@@ -238,22 +238,21 @@ ObjExtend(struct worker *wrk, struct objcore *oc, ssize_t l)
 uint64_t
 ObjWaitExtend(const struct worker *wrk, const struct objcore *oc, uint64_t l)
 {
-	uint64_t rv;
-
 	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
 	CHECK_OBJ_NOTNULL(oc, OBJCORE_MAGIC);
 	CHECK_OBJ_NOTNULL(oc->boc, BOC_MAGIC);
-	Lck_Lock(&oc->boc->mtx);
+
+	Lck_AssertHeld(&oc->boc->mtx);
+
 	while (1) {
-		rv = oc->boc->len_so_far;
-		assert(l <= rv || oc->boc->state == BOS_FAILED);
-		if (rv > l || oc->boc->state >= BOS_FINISHED)
+		assert(l <= oc->boc->len_so_far ||
```

Comment:
I had reviewed the original code just recently and one of the results I remember was that the consistent view through the rv local variable was important for correctness.
So here I'd like to turn the argumentation around: What would be the good reason to change it?